client/lxd_instances.go

```diff
@@ -83,6 +83,31 @@ func (r *ProtocolLXD) GetInstances(instanceType api.InstanceType) ([]api.Instanc
 	return instances, nil
 }
 
+// GetInstancesAllProjects returns a list of instances from all projects.
+func (r *ProtocolLXD) GetInstancesAllProjects(instanceType api.InstanceType) ([]api.Instance, error) {
+	instances := []api.Instance{}
+
+	path, v, err := r.instanceTypeToPath(instanceType)
+	if err != nil {
+		return nil, err
+	}
+
+	v.Set("recursion", "1")
+	v.Set("all-projects", "true")
```

Comment:
Ah, that's an interesting approach instead of just adding another level of recursion :)

I think that makes sense, though then we should probably also have a `GetInstanceNamesAllProject` which gives us a `map[string][]string`

Reply:
@stgraber can you explain more deeply why we need `GetInstanceNamesAllProject` function and where we should use it?

Reply:
It's just for consistency. We won't use it from within our own CLI but that doesn't mean that others relying on our Go client package won't have a use for it.